### PR TITLE
Rework for the Navbar

### LINKED ISSUE #258: Refactor Nav Bar to properly scale as we add more features

### PR DESCRIPTION
**Pull request by JoshDevHub**
## What's the change?
- profile picture tag helper
- New navigation system! The building blocks of this are:
  - a `NavigationContainerComponent`: this generates the markup + styles for the new navigation layout. 'Primary Links' are oriented horizontally along the navbar at desktop view width and turn into a 'drawer' at mobile width. Account-oriented links are in a dropdown on the right side of the navbar for both desktop and mobile widths. This right side button becomes a plain 'Login' button if the user is not logged in. (see demo screenshots for probably better illustrations of this)
  - a `NavigationComponent`: this component outlines the template for how its subclasses should use the `NavigationContainterComponent`. It also uses its `.for` method to determine which of its subclasses should be used for any given user.
  - `Navigation::{UserRole}Component`: there's one of these for each type of user. Because each user needs a different set of navigational links, this felt like a natural way to do this while having it easy to understand what links appear for which user and easy to change going forward. There are arguably some issues with this design, like the same links getting repeated for different users (ie. every logged in user names a 'Logout' link). I think this is fairly low cost repetition but glad to discuss more if people want.

## What key workflows are impacted?
Using the navbar

#### Desktop Navbar for Member Users
Selection_310

#### Desktop Navbar with Profile Dropdown Open
Selection_311

#### Mobile View (no menus open)
Selection_312

#### Mobile View with Drawer Open
Selection_315

#### Mobile View with Dropdown Open
Selection_316

## Issue ticket number and link
Closes #258 

## Checklist before requesting a review

Please delete items that are not relevant.

- [x] Did you add appropriate automated tests?
- [x] Did you consider risks around security, performance, etc.?
- [x] Have you thought of misfiring code? e.g. too many loops, n+1, or how to handle nils?
- [x] Related to readability and consistency: Did you add comments and/or documentation? (README, Notion, etc.)
- [x] Did you leave helpful inline PR comments for the reviewer(s)?
- [x] Did you include instructions for how to test in the ticket?
- [x] Did you add any relevant tags and decide if this PR is a Draft vs. Ready for Review?

Definitely suggest reviewer(s) of this checkout this PR's branch and login to different users and check things out. Difficult to fit every single scenario into the demo/screenshot section. Also good to know how it feels to use this.

### CODE REVIEW

**Review by toyhammered — COMMENTED** [review]
Holy hell I love this! I think this will work well for now (as long as we don't get too crazy with the navbar links).

File: app/components/navigation_component.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class NavigationComponent < ViewComponent::Base
+  include ProfilesHelper
+
+  def self.for(current_user:)
+    role = (current_user&.role || 'Guest').capitalize
+    subclass = "Navigation::#{role}Component".safe_constantize
```
Comment (afogel) [inline app/components/navigation_component.rb:8]: slick 😎

File: app/components/navigation_container_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            <% end %>
+          </ul>
+        </div>
+      </div>
+      <% if render_dropdown? %>
+        <div class="dropdown dropdown-end dropdown-bottom flex">
+          <label tabindex="0" class="cursor-pointer px-6">
+            <%= profile_picture %>
+          </label>
+          <ul tabindex="0" class="menu menu-compact dropdown-content mt-3 shadow bg-base-300 rounded-box w-52">
```
Comment (afogel) [inline app/components/navigation_container_component.html.erb:25]: can we make the padding top and padding bottom be a bit larger for the first and last links  (respectively) in the dropdown?

**Review by afogel — COMMENTED** [review]
overall, first pass looked 👌 , just had one small request. Will do a second pass later, but thanks for taking the initiative and bringing this to fruition!

**Comment by JoshDevHub** [comment]
Marking this ready for review. I thought I had some other changes to make around hover states with the nav links on desktop, but turns out my monitor was just being weird. Only real changes I've made are based around better spacing in the dropdown (shoutouts to Ariel for these ideas)

#### Before:
270484381-db003bd3-1371-4035-b8e7-85851f473b2a

#### After:
Selection_320

**Review by afogel — APPROVED** [review]
lgtm, great job.

**Pull request merged**